application/src/main/java/org/opentripplanner/routing/algorithm/raptoradapter/transit/request/TripTimesForDaysIndex.java

```diff
@@ -0,0 +1,130 @@
+package org.opentripplanner.routing.algorithm.raptoradapter.transit.request;
+
+import java.util.List;
+
+/**
+ * This class build an tripIndex for {@link TripPatternForDates} to (day, tripIndexForDay).
+ * In most cases concatenating the depature times would be correct, but the last depature on a
+ * specific day, may depart after the first departure on the following day. In these cases we need
+ * to swap the two depatures in the timetable Raptor searches. The reason we need this, is that
+ * trips on two diffrent datys may have overlapping trip-times. A nightbus on Saturdays may leave
+ * at 04:05+1d, while the first bus on Sundays may leave at 03:00. Be aware, these two trips may
+ * not have the same service calendar, normally they are diffrent. For example "Sundays" might be
+ * all Sundays and public holidays.
+ * </p>
+ * The trip index is a sorted list of trips based on the first stop depature time. The index
+ * contains two pointers, the first is the day and the second is the trip index on that day:
+ * ```
+ *   tripIndexForTripPatternPerDates -> (day, tripIndexForTripPatternPerDate)
+ * ```
+ * </p>
+ * This class might at first look a bit complicated. A much easier approch would be to just sort
+ * on depature times using a Java build in sort, but sorting is expencive. This code will merge
+ * the  timetables instead - witch is faster. In 99% of the cases we will just do one extra
+ * comparasons for each depature time.
+ */
+final class TripTimesForDaysIndex {
+
+  private final int[] tripIndex;
+
+  static TripTimesForDaysIndex ofTripTimesForDay(List<int[]> departureTimes, int[] offsets) {
+    departureTimes = applyOffsets(departureTimes, offsets);
+    return new TripTimesForDaysIndex(departureTimes);
+  }
+
+  TripTimesForDaysIndex(List<int[]> firstStopDepartureTimesPerDay) {
```

Comment:
I think that this should have a Javadoc comment with a short explanation of the algorithm because the logic is not easily readable

Reply:
I will add more doc to this.